WCGitTagsPlugin/WCGitTagsPlugin.m

```diff
@@ -65,39 +65,50 @@ - (id)initWithBundle:(NSBundle *)plugin
         // reference to plugin's bundle, for resource acccess
         self.bundle = plugin;
         
-        NSMenuItem *menuItem = [[NSApp mainMenu] itemWithTitle:@"Source Control"];
-        if (menuItem) {
-            
-            NSMenuItem *actionMenuItem = [[NSMenuItem alloc] initWithTitle:@"Tags..." action:@selector(presentTagsModal:) keyEquivalent:@""];
-            [actionMenuItem setTarget:self];
-            
-            self.refreshStatusItem = [[menuItem submenu] itemWithTitle:@"Refresh Status"];
-            [self.refreshStatusItem addObserver:self forKeyPath:@"enabled" options:0 context:NULL];
-            
-            NSInteger indexOfRefreshStatusItem = [[menuItem submenu] indexOfItem:self.refreshStatusItem];
-            if (indexOfRefreshStatusItem == -1) {
-                [[menuItem submenu] addItem:[NSMenuItem separatorItem]];
-                [[menuItem submenu] addItem:actionMenuItem];
-            } else {
-                [[menuItem submenu] insertItem:actionMenuItem atIndex:indexOfRefreshStatusItem];
-            }
-            
-            self.tagsItem = actionMenuItem;
-            [self.tagsItem setEnabled:self.refreshStatusItem.isEnabled];
-            
-            WCTagWatchdog *watchDog = [[WCTagWatchdog alloc] initWithWatchBlock:^{
-                [self willChangeValueForKey:@"tags"];
-                [self didChangeValueForKey:@"tags"];
-            }];
-            watchDog.gitDirectoryURL = self.repository.gitDirectoryURL;
-            self.watchDog = watchDog;
-            
-            self.beingPresented = NO;
-        }
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(didApplicationFinishLaunchingNotification:)
+                                                     name:NSApplicationDidFinishLaunchingNotification
+                                                   object:nil];
+        
+
+        WCTagWatchdog *watchDog = [[WCTagWatchdog alloc] initWithWatchBlock:^{
+            [self willChangeValueForKey:@"tags"];
```

Comment:
I know this is copy-paste of my code, but there's a memory leak here. `self` should be weak.